### ISSUE TITLE
Floating point exception(core dumped)

### ISSUE THREAD

**Issue by amin-saffari** [champost/ABLE#23]
Hi Champak,

Sometimes I have got "Floating point exception(core dumped)". To be more informative, I ran gdb and Valgrind on ABLE, and they point me to 
```C
if (quo) {
    rem = quo % mutClass;
    quo /= mutClass;
}
```
[239-240](https://github.com/champost/ABLE/blob/master/main.cpp#L239) and [261-262](https://github.com/champost/ABLE/blob/master/main.cpp#L261). 
For an unknown reason (to me), if I enter these two functions with mutClass=zero, given the C++ Standard(2003) in §5.6/4 the **behaviour** in both cases (divide and modulo) is **undefined**.

Alternatively, I try to tame the "behaviour" by conditioning on both ```quo && mutClass``` ( [238](https://github.com/champost/ABLE/blob/master/main.cpp#L238) and [260](https://github.com/champost/ABLE/blob/master/main.cpp#L260)). However, I am still facing "Floating point exception(core dumped)", Do you have any other suspicious places in mind, where I should explore?

Cheers,
Amin

**Comment by champost** [comment]
HI Amin,
What you describe appears to be a case of _division by zero_ and I could improve the checks for such cases but could you tell me in what situations does this occur, _e.g._ contents of your config file and command line ?

**Comment by amin-saffari** [comment]
Hi Champak,

As I have mentioned and you also pointed out, this is for sure an arithmetic exception. However, it is not clear to me why mutClass enters those two functions with its initial value!?

Here is the content of my config file: 
```
datafile /scr/k71/rohit/ABLE/data/sparse.1000.90.bSFS
datafile_format bSFS
task infer
folded
kmax 3

pops 2 2 2

start random
global_search CRS
global_search_trees 10000
local_search_trees 20000

retain_best_points 300
global_search_iterations 3
global_bounds_refinement progressive

bounds tbi1 1e-3 5 0 21
bounds tbi2 1e-3 5 0 20
bounds tbi3 1e-3 5 0 30
bounds tbi4 1e-3 5 0 30
bounds tbi5 1e-3 5 0 30
bounds tbi6 1e-3 0.5 0 1
bounds tbi7 1e-3 0.5 0 1
bounds tbi8 1e-3 5 0 30

constrain tbi8 tbi3

refine_likelihoods 10000
report_likelihoods 10000

set_threads 30
```
And my ABLE command:

```ABLE 4 xxx -t tbi1 -r tbi2 1001 -I 2 2 2 -n 1 tbi4 -n 2 tbi5 -es tbi8 1 tbi6 -ej tbi8 3 2 -es tbi8 2 tbi7 -ej tbi8 4 1 -ej tbi3 1 2```

Thanks for your prompt support.

Cheers,
Amin

**Comment by champost** [comment]
Hi Amin,
I looked at your issue but I could not reproduce the division by zero exception. As to your question regarding why `mutClass` would be zero, it depends on how `kmax` is being set by the user in the config file. If the user doesn't specify a value for `kmax` then by default it is calculated by the data supplied by the user. When no data needs to be specified (_i.e._ `task exact_bSFS`) then `ABLE` should spit out an error and aborts.

It could be that a previous commit resulted in your exception so if you know the commit hash it would be very helpful. I have also started labelling my commits in terms of a version number to make it easier for users to report a problem in the future (#24 ).

Cheers,
Champak
